### ISSUE TITLE
CQL2/Grammar: Quoted identifiers not defined as expected

### ISSUE THREAD

**Issue by jerstlouis** [opengeospatial/ogcapi-features#916]
My understanding is we should be able to put just about anything in quoted identifiers (`" "`).

Currently, the `propertyName` rule which is the closest thing that corresponds to an identifier token uses the exact same `identifier` inside and outside the quote:

```
propertyName = identifier | "\"" identifier "\"";
```

I seem to recall us discussing this before, but don't recall the decision.
In the currently state, this completely defeats the purpose of the quoted identifiers however.

A use case in our CartoSym-CSS styles is a selector on the [`sentinel2-l2a`](https://maps.gnosis.earth/ogcapi/collections/sentinel2-l2a) layer which contains a hyphen.

See how we defined this in the CartoSym-CSS BNF for ANTLR:

https://github.com/opengeospatial/styles-and-symbology/blob/main/core/schemas/CartoSym-CSS-Lexer.g4#L110

allowing inside the double quotes all the same characters as inside a string literal with the exception of the double quote.

The only thing that this might currently be missing the the capability to backslash escape the double quote `\"` or to otherwise allow to use single quotes directly).

**Comment by cportele** [comment]
Meeting 2024-04-08: The original issue was #332 which included the following statement:

> GeoServer CQL implementation supports double quoted identifiers for this very reason, avoid clashes and deal with special characters that might have been used a part of an identifier (relationa databases seems to allow pretty much anything in column names, as long as the name is double quoted).

The PR that closed this issue (in 2020) resolved the "avoid clashes" aspect, but not the "deal with special characters in an identifier" part. The text in #332 is not entirely clear if this was intended or not. Since this change is also in the BNF for years it is unclear, if we could change this now as the document is now under TC control. 

@pvretano, any thoughts?

**Comment by cportele** [comment]
@jerstlouis - I don't think there is a problem in our grammar (or the BNF) wrt to avoiding clashes. We test all the different cases in unit tests and the results are as expected.

If the parser reads `S_INTERSECTS(bbox,BBOX(0,0,180,90))` it fails since the parser expects a `(` after `bbox`. If the parser reads `S_INTERSECTS("bbox",BBOX(0,0,180,90))` it will parse `"bbox"` as a `propertyName` and the property identifier is `bbox`.

**Comment by jerstlouis** [comment]
@cportele right, sorry, I just tried that with your lexer and parser on http://lab.antlr.org/ . Thanks for double-checking.

I'm still confused as to why there's no error or ambiguity given that `IdentifierBare` is not marked as a fragment (things still work the same marking it as such).

In the BNF grammar as a single piece though (not separated as lexer + parser), I thought it might be ambiguous, but again perhaps it gets resolved correctly with `" Identifier "` being the only possible resolution since `" BBOX "` does not occur anywhere.

**Comment by cportele** [comment]
Meeting 2024-04-22: The current rule is what it is since December 2020 and it did not raise any issues in implementations. Given that the document is under TC control now (in the approval vote) and this is not a bug, we cannot make the change now. It would either require a TC vote with a comment or it is future work.

**Comment by cportele** [comment]
Since no issue was raised in the approval process, I move this issue to "future work" (see the previous comment).
